### PR TITLE
Adding a base docker image for builds using gcc

### PR DESCRIPTION
**Pull request by lranjbar**
Similar to the go base image that @ernestojeda made. Goal is to have a bare bones image for the gcc tool chain.

Signed-off-by: Lisa Rashidi-Ranjbar <lisa.a.rashidi-ranjbar@intel.com>

### CODE REVIEW

**Comment by jamesrgregg** [comment]
@lranjbar when I was testing the build of device-sdk-c I also needed these additional dependencies
wget libcurl 
Can we add them as well?

**Comment by lranjbar** [comment]
@jamesrgregg @MightyNerdEric I'm still testing these changes on sandbox this was more to test out squashing commits.

**Comment by lranjbar** [comment]
Dependency on #282

**Review by ernestojeda — COMMENTED** [review]
I am not sure I follow why all these custom triggers are needed now. I think I may be missing some context. I thought the goal was to remove any reverse triggers for '{project-name}-{stream}-merge-java' except in the case of the `support-rulesengine.yaml` template. Why is it needed in so many non-java projects like `edgex-mongo.yaml` or `core-consul.yaml`

File: jjb/core/core-consul.yaml [hunk 1]
```diff
@@ -19,18 +19,30 @@
           docker_tag: 'master'
     jobs:
       - '{project-name}-{stream}-verify-docker'
-      - '{project-name}-{stream}-merge-docker'
-      - '{project-name}-{stream}-release-version-docker-daily-no-sonar'
+      - '{project-name}-{stream}-merge-docker':
+          custom_triggers:
+            - reverse:
+                jobs: '{project-name}-{stream}-merge-java'
```
Comment (ernestojeda) [inline jjb/core/core-consul.yaml]: See my review comment for context. Not following why this is needed, I thought we were removing merge-java references.

File: jjb/ci-management/docker.yaml [hunk 2]
```diff
@@ -16,11 +16,19 @@
           docker_tag: 'master'
     jobs:
       - '{project-name}-{stream}-verify-docker'
-      - '{project-name}-{stream}-merge-docker'
+      - '{project-name}-{stream}-merge-docker':
+          custom_triggers:
+            - github
```
Comment (ernestojeda) [inline jjb/ci-management/docker.yaml]: Is this a new trigger? Should we be adding this to other docker templates.

**Comment by lranjbar** [comment]
@ernestojeda @JPWKU @jamesrgregg 

Just rebased this change and it should be good to go now with the other PRs resolved.

**Review by eb-oss — APPROVED** [review]

**Review by eb-oss — CHANGES_REQUESTED** [review]
Please change commit message to be imperative ("Add a base docker image...").

**Review by jamesrgregg — APPROVED** [review]
LGTM - thanks for fixing the commit message

**Review by eb-oss — APPROVED** [review]

**Pull request merged**